internal/reconciler/metricpipeline/status.go

```diff
@@ -77,6 +85,49 @@ func (r *Reconciler) updateStatus(ctx context.Context, pipelineName string, lock
 	return setCondition(ctx, r.Client, &pipeline, pending)
 }
 
+func (r *Reconciler) updateAgentStatus(ctx context.Context, pipelineName string) error {
```

Comment:
I think it's good that you made `updateAgentStatus` its own function. How about doing the same with `updateGatewayStatus`? This way, we might not need `updateStatusConditions` and can do everything in `updateStatus` directly.

Reply:
I need `updateStatusConditions` since `updateAgentStatus` depend on result of previous checks, putting all in a single function will make code hard to read

Reply:
gateway and secret references are checked in the funtion